src/script/interpreter.cpp

```diff
@@ -1830,6 +1830,19 @@ uint256 ComputeTapleafHash(uint8_t leaf_version, Span<const unsigned char> scrip
     return (HashWriter{HASHER_TAPLEAF} << leaf_version << CompactSizeWriter(script.size()) << script).GetSHA256();
 }
 
+uint256 ComputeTapbranchHash(Span<const unsigned char> a, Span<const unsigned char> b)
+{
+    assert(a.size() == 32);
+    assert(b.size() == 32);
```

Comment:
Presumably we want to stop if we are misusing the API somewhere, rather than proceeding with erroneous computations?  In particular these assertions were a cheap way of helping ensure that the invariant that `ComputeTapbranchHash` only ever uses its tagged hash on tapbranch data.  Normally I'd use `uint256` arguments here (and perhaps I should), but that would involve extra copying in some cases.

Anyhow, I don't feel strongly, so I've removed the assertions.